src/Components/Header/Header.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useContext } from 'react';
+
+import { Context } from '../../Context';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { Todo } from '../../types/Todo';
+import { addTodo } from '../../api/todos';
+
+export const Header = () => {
+  const [query, setQuery] = useState('');
+  const {
+    USER_ID,
+    handleErrorChange,
+    completeAll,
+    handleActiveTodos,
+  } = useContext(Context);
+
+  const handleQueryChange = (value: string) => {
+    handleErrorChange('');
+    setQuery(value);
+  };
+
+  const handleAddTodo = (todoTitle: string) => {
+    const newTodo: Omit<Todo, 'id'> = {
+      title: todoTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    addTodo(newTodo)
+      .catch(() => handleErrorChange(ErrorMessage.UNABLE_TO_ADD));
+  };
+
+  const addingTodo = (event:React.FormEvent<HTMLFormElement>) => {
```

Comment:
No need to split the logic into 2 functions - write one. You could move the validation into a separate function, but it's very small, so there is no necessary to do it